v1.3: add ranking rules details (#2476)
meilisearch/documentation: reference/api/search.mdx

<CodeSamples id="search_parameter_guide_matching_strategy_2" />

The above code sample would only return documents containing all three words.

### Ranking score

**Parameter**: `showRankingScore`<br />
**Expected value**: `true` or `false`<br />
**Default value**: `false`

> Adds a global ranking score field, `_rankingScore`, to each document. The `_rankingScore` is  between `0` and `1`. The higher the `_rankingScore`, the higher the document will rank.

Reviewer: ```suggestion
Adds a global ranking score field, `_rankingScore`, to each document. The `_rankingScore` is a numeric value between `0.0` and `1.0`. The higher the `_rankingScore`, the more relevant the document.
```

> The `sort` ranking rule does not influence the `_rankingScore`. Instead, the document order is determined by the value of the field they are sorted on. Excluding `sort`, if two documents have the same `_rankingScore`, they are returned in an unspecified order.

Author: >. What do we mean with "unspecified order"?

If docA and docB both have a `_rankingScore` of 0.39, the order in which they are returned is unspecified. Sometimes I may get docA first, sometimes docB.

I think we should keep this sentence so users know why the order of some results "changes" sometimes.

>Have you confirmed with the devs this behaviour is truly unspecified?

I did. [Link to Slack thread](https://meilisearch.slack.com/archives/C05A9SRPMFB/p1687253581233329)

> The `sort` ranking rule does not influence the `_rankingScore`. Instead, the document order is determined by the value of the field they are sorted on. Excluding `sort`, if two documents have the same `_rankingScore`, they are returned in an unspecified order.

Reviewer: ok, thanks for the link. I agree with @dureuill: we shouldn't document this. Removing the last sentence is probably the best way of doing it.

To say anything about it, like stating the behaviour is unspecified, will only raise questions we can't really answer. Not only we can't answer, but at the moment there seems to be no case where this situation might significantly impact users.
